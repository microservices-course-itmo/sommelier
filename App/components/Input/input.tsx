import React, { useState, useCallback } from 'react'
import { TextInput } from 'react-native'

import * as Styled from './input.styled'

import Ruble from '../../../assets/images/ruble.svg'

type Props = {
  placeholder?: string
  value?: string
}

const InputPrice = ({ placeholder, value = '' }: Props) => {
  const [isFocused, setIsFocused] = useState(false)
  const [text, setText] = useState(value)

  const inputRef = React.createRef<TextInput>()

  const handleInputFocus = useCallback(() => {
    setIsFocused(true)
  }, [setIsFocused])

  const handleInputBlur = useCallback(() => {
    setIsFocused(false)
  }, [setIsFocused])

  const toggleFocus = useCallback(() => {
    setIsFocused(!isFocused)
  }, [isFocused, setIsFocused])

  const textInput = useCallback(
    (text) => {
      setText(text)
    },
    [text, setText]
  )

  const isInputEmpty = text !== ''

  return (
    <Styled.StyledContainer>
      <Styled.StyledPlaceholderLabel
        isFocused={isFocused}
        isFilled={isInputEmpty}
        onPress={toggleFocus}
      >
        {placeholder}
      </Styled.StyledPlaceholderLabel>
      <Styled.StyledLabeledInput
        keyboardType='numeric'
        placeholderTextColor='#DCDCDC'
        onFocus={handleInputFocus}
        onBlur={handleInputBlur}
        onChangeText={textInput}
        value={text}
        ref={inputRef}
        data-test='sommelier-input-price'
      />
      <Styled.StyledRubleIcon>
        <Ruble width='12px' height='21px' />
      </Styled.StyledRubleIcon>
    </Styled.StyledContainer>
  )
}

export default InputPrice
