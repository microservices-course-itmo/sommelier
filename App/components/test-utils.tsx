import React from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const themeRender = (component: any) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
