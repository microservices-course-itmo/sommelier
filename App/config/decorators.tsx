import { ThemeProvider } from 'styled-components'
import React from 'react'
import styled from 'styled-components/native'
import { theme } from '../theme'

const StyledBufferView = styled.View`
  flex: 1;
  margin-vertical: 40;
  margin-horizontal: 20;
`

export const ThemeDecorator = (storyFn: any) => (
  <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
)

export const BufferView = (storyFn: any) => (
  <StyledBufferView>{storyFn()}</StyledBufferView>
)
