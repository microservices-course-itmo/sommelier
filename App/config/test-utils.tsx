import React, { ReactNode } from 'react'
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../theme'

const themeRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}

export * from '@testing-library/react-native'

export { themeRender }
