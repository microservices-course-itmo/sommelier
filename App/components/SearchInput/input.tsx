import React, { useState } from 'react'
import {
  NativeSyntheticEvent,
  TextInputChangeEventData,
  TouchableOpacity,
} from 'react-native'
import theme from '../../theme/index'
import * as Styled from './input.styled'
import Search from '../../../assets/images/search.svg'
import Close from '../../../assets/images/close.svg'
import { INPUT_SIZE_KEYS } from '../../theme/inputSearchSizes'

type Props = {
  defaultValue?: string
  placeholder?: string
  editable?: boolean
  size?: keyof typeof theme.inputSearchSizes
  maxLength?: number
  value?: string
  onChange: (e: NativeSyntheticEvent<TextInputChangeEventData>) => void
  clearButtonMode?: boolean
}

const SearchInput = ({
  defaultValue,
  placeholder,
  editable = true,
  size = INPUT_SIZE_KEYS.md,
  value,
  onChange,
  clearButtonMode,
  maxLength,
}: Props) => {
  const [text, setText] = useState(value)
  return (
    <Styled.StyledView editable={editable}>
      <Search
        width={theme.searchSizes[size]}
        height={theme.searchSizes[size]}
      />
      <Styled.StyledInput
        defaultValue={defaultValue}
        placeholder={placeholder}
        editable={editable}
        size={size}
        value={text}
        onChange={onChange}
        onChangeText={(text) => setText(text)}
        maxLength={maxLength}
      />
      {clearButtonMode && (
        <Styled.ViewClose>
          <TouchableOpacity onPress={() => setText('')}>
            <Close
              width={theme.inputSearchFontSizes[size]}
              height={theme.inputSearchFontSizes[size]}
            />
          </TouchableOpacity>
        </Styled.ViewClose>
      )}
    </Styled.StyledView>
  )
}

export default SearchInput
