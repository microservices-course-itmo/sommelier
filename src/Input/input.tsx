import React, { ReactElement, ReactNode } from 'react'
import * as Styled from './input.styled';
import { SIZE } from '../theme/spacing';


type Props = {
    name?: string;
    defaultValue?: string;
    placeholder?: string;
    disabled?: boolean;
    prefix?: ReactNode;
    size?: keyof typeof SIZE;
    maxLength?: number;
    value?: string;
    onChange?: (inputValue: string) => void;
    allowClear?: boolean;
}

export const Input = ({ size = SIZE.md, maxLength, value, onChange, allowClear }: Props): ReactElement => (
    <Styled.StyledInput size={size}></Styled.StyledInput >
);
