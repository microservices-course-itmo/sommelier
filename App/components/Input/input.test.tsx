import React from 'react'
import { themeRender } from '../../config/test-utils'
import { InputPrice } from './input'

describe('Input Price component', () => {
  it('Component with only necessary props matches snapshot', () => {
    const component = themeRender(<InputPrice />)
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('Component with all props matches snapshot', () => {
    const component = themeRender(<InputPrice placeholder='123' value='123' />)
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('Placeholder prop assigns correctly', () => {
    const component = themeRender(<InputPrice placeholder='123' />)
    expect(
      component.getByTestId('sommelier-input-price-placeholder').children[0]
    ).toBe('123')
  })
  it('Value prop assigns correctly', () => {
    const component = themeRender(<InputPrice value='123' />)
    expect(
      component.getByTestId('sommelier-input-price-label').props.value
    ).toBe('123')
  })
})
