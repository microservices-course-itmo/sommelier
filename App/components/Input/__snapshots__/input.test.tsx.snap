// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Price component Component with all props matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#dcdcdc",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "marginTop": 40,
        "paddingLeft": 1,
        "paddingRight": 12,
        "width": 126,
      },
    ]
  }
>
  <Text
    isFilled={true}
    isFocused={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#B5B5B5",
          "fontSize": 12,
          "left": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 4,
            },
          ],
        },
      ]
    }
    testID="sommelier-input-price-placeholder"
  >
    123
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholderTextColor="#DCDCDC"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#333333",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 18,
          "marginBottom": -20,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
    testID="sommelier-input-price-label"
    underlineColorAndroid="transparent"
    value="123"
  />
  <View
    style={
      Array [
        Object {
          "bottom": 2,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
  >
    <SvgMock
      height="21px"
      width="12px"
    />
  </View>
</View>
`;

exports[`Input Price component Component with only necessary props matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#dcdcdc",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "height": 50,
        "marginTop": 40,
        "paddingLeft": 1,
        "paddingRight": 12,
        "width": 126,
      },
    ]
  }
>
  <Text
    isFilled={false}
    isFocused={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#B5B5B5",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 25,
            },
          ],
        },
      ]
    }
    testID="sommelier-input-price-placeholder"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholderTextColor="#DCDCDC"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "color": "#333333",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 18,
          "marginBottom": -20,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
    testID="sommelier-input-price-label"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Array [
        Object {
          "bottom": 2,
          "position": "absolute",
          "right": 0,
        },
      ]
    }
  >
    <SvgMock
      height="21px"
      width="12px"
    />
  </View>
</View>
`;
